src/controller/python/chip/ChipDeviceCtrl.py

```diff
@@ -290,6 +290,29 @@ def CheckTestCommissionerCallbacks(self):
     def CheckTestCommissionerPaseConnection(self, nodeid):
         return self._dmLib.pychip_TestPaseConnection(nodeid)
 
+    def CommissionWithCode(self, setupPayload: str, nodeid: int):
+        self.CheckIsActive()
+
+        setupPayload = setupPayload.encode() + b'\0'
+
+        # IP connection will run through full commissioning, so we need to wait
+        # for the commissioning complete event, not just any callback.
+        self.state = DCState.RENDEZVOUS_ONGOING
+
+        self._ChipStack.commissioningCompleteEvent.clear()
+
+        self._ChipStack.CallAsync(
+            lambda: self._dmLib.pychip_DeviceController_ConnectWithCode(
+                self.devCtrl, setupPayload, nodeid)
+        )
+        # Wait up to 5 additional seconds for the commissioning complete event
```

Comment:
This is not long enough. Why 5 seconds? Why not base it on the sleepy interval, etc ?